libraries/iio/Makefile

```diff
@@ -0,0 +1,56 @@
+ifeq ($(OS), Windows_NT)
+remove_fun = del /S /Q $(subst /,\,$1)
+else
+remove_fun = rm -rf $(1)
+endif
+
+# Local variables
+TINYIIOD_VERSION_MAJOR = 0
+TINYIIOD_VERSION_MINOR = 1
+TINYIIOD_VERSION = $(TINYIIOD_VERSION_MAJOR).$(TINYIIOD_VERSION_MINOR)
+
+BUFF_SIZE ?= 0x1000
+STD_TYPES ?= true
+
+# Build settings
+IIO_DEFS = -D TINYIIOD_VERSION_MAJOR=$(TINYIIOD_VERSION_MAJOR)		\
+	   -D TINYIIOD_VERSION_MINOR=$(TINYIIOD_VERSION_MINOR)		\
+	   -D TINYIIOD_VERSION_GIT=0x$(shell git rev-parse --short HEAD)\
+	   -D IIOD_BUFFER_SIZE=$(BUFF_SIZE)
+
+ifeq (true,$(strip $(STD_TYPES)))
+IIO_DEFS += -D_USE_STD_INT_TYPES
+endif
+
+IIO_DIR		= $(realpath .)
+TINYIIOD_DIR	= $(IIO_DIR)/libtinyiiod
+
+LIB_NAME	= libiio.a
+
+SRCS = iio.c $(TINYIIOD_DIR)/parser.c $(TINYIIOD_DIR)/tinyiiod.c
+
+OBJS = $(SRCS:.c=.o)
+
+CFLAGS += $(IIO_DEFS) \
+	  -I"$(TINYIIOD_DIR)"
+
+.PHONEY = all clean
+
+#Init tiniiod submodule if not initialized
+ifeq ($(wildcard $(TINYIIOD_DIR)/README.md),)
```

Comment:
Wildcards are used for a specific pattern, not for a single file

Reply:
In this case wildcard is used to check if file exists.

Reply:
> In this case wildcard is used to check if file exists.

You can use this as an example

FILE=/etc/resolv.conf
if [ -f "$FILE" ]; then
    echo "$FILE exists."
else 
    echo "$FILE does not exist."
fi

Also wouldn't assume if a file named README.md exists in a folder, it world be a repository

Reply:
I think is simpler this way.
I didn't find an other way to check if the submodule is initialized.

Reply:
Also it seems what you propose is a shell scrips, which I would rather avoid inside the makefile

Reply:
You could check for the .git folder. This might be more generic.

Reply:
will solve here: https://github.com/analogdevicesinc/no-OS/pull/674